Feat(/#265):- Back to top component. (#104)
1-Platform/op-components: packages/opc-back-to-top/README.md

> # This is opc-back-to-top 👋

Reviewer: rename to `This is opc-back-top-top component`

Reviewer 2: ```suggestion
# This is opc-back-to-top  component 👋
```
@anjneeksharma  What @diwanshi mentioned (-top-top) must be a typo. I suppose she meant what I have suggested

Reviewer: yes that was a typo on my end

![Version](https://img.shields.io/badge/version-0.0.1-blue.svg?cacheSeconds=2592000)
![Build Status](https://travis-ci.org/dwyl/esta.svg?branch=master)
[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/1-Platform/op-components/

A standardized web component based on Lit element for Red Hat One Platform back to top.

## Install

```sh
npm install
```

## Usage

> plain back to top

Reviewer: rename to `Default back-to-top`

```html

> <opc-back-to-top></opc-back-to-top>

Reviewer: @anjneeksharma this Readme doesn't follow the standard Readme format that we have created. Please update it accrodingly

Reviewer: all the attributes and slots should be mentioned separately

Author: @diwanshi i am not able to understand your this comment, can you please elaborate bit ?

Reviewer: @anjneeksharma can u use the standard readme format that we are using in opc-header component

```
Back-to-top with reveal-at attribute, It reveal back to top once user scroll upt to specified px down.
```html
<opc-back-to-top reveal-at="200"></opc-back-to-top>
```

> Back to top with slot, In slot user can pass its own choice icon and text.

Reviewer 3: ```suggestion
Back to top with slot, In slot users can pass icon and text of their choice.
```

> Back to top with slot, In slot users can pass icon and text of their choice.

Reviewer: this should be written as `back-top-top with default slot (optional) where users can pass icon and text if they like`

[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/1-Platform/op-components/graphs/commit-activity)

A standardized web component based on Lit element for Red Hat One Platform back to top.

## Usage
Default back-to-top
```html
<opc-back-to-top></opc-back-to-top>
```
- Back-to-top with reveal-at attribute, It reveals back to top once user scrolls upto specified pixels downwards.
```html
<opc-back-to-top reveal-at="200"></opc-back-to-top>
```

- Back-to-top with reveal-at attribute, It reveals back to top once user scrolls upto specified pixels downwards.
```html
<opc-back-to-top reveal-at="200"></opc-back-to-top>
```

```html
<opc-back-to-top></opc-back-to-top>
```

## Slots
- Back-to-top with default slot (optional) where users can pass icon and text if they like.
```html
<opc-back-to-top>
  <button class="rh-text" tabindex="0">
    <svg xmlns='http://www.w3.org/2000/svg' width='20px' height='20px' viewBox='0 0 512 512'>
    <title>ionicons-v5-a</title>
    <polyline points='112 328 256 184 400 328' style='fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:48px'/>
    </svg>

>     Go-up

Reviewer: rename this to `Go to Top`

  </button>
</opc-back-to-top>
```

## Attributes

> <!-- Add attributes here -->

Reviewer: @anjneeksharma can u please add the attributes here pls

```sh
npm install
```

## Usage

### Install opc-back-to-top

```sh
npm install --save @one-platform/opc-back-to-top 
```

### For VanillaJS
- Import component
```js
import '@one-platform/opc-back-to-top';
```
- Add component in html
```html
<opc-back-to-top>
</opc-back-to-top>
```

### For Angular
- In your app.module include the `CUSTOM_ELEMENTS_SCHEMA` and import the component
```js
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import '@one-platform/opc-back-to-top';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
  ],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```
- Add component in any component html template
```html
<opc-back-to-top>
</opc-back-to-top>
```

### For React
- Import the component in App.js
```js
import '@one-platform/opc-back-to-top';
```

- Add component in any component html render
```html
<opc-back-to-top>
</opc-back-to-top>
```

### Development server

- Run development server

```sh
npm run dev opc-back-to-top
```

### Build

```sh
npm run build opc-back-to-top
```

## Run tests

```sh
npm run test
```

## 🤝 Contributors

> 👤 **anjnee**

Reviewer: Add a proper name here with email.
